### PR TITLE
Switch Travis CI to Trusty Beta

### PR DESCRIPTION
**Pull request by zpao**
Trying out https://docs.travis-ci.com/user/trusty-ci-environment to see if we can make the website build again.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 ---
+sudo: required
+dist: trusty
 language: node_js
 node_js:
 - 4
-sudo: false
+rmv:
```
Comment (alexzherdev) [inline .travis.yml]: Typo? (rvm)

**Comment by spicyj** [comment]
(The website doesn't build?) Why sudo required?

**Comment by krzysiek1507** [comment]
@spicyj https://docs.travis-ci.com/user/trusty-ci-environment/#tl%3Bdr---Using-Trusty

**Comment by spicyj** [comment]
Woah GCE. ¯\_(ツ)_/¯  okay.

**Comment by zpao** [comment]
sudo is required for trusty.

The website doesn't build currently because the version of Ruby installed is old and I'm pretty sure it's a pain to upgrade (the images are configured primarily based on the `language` field but not set up great to handle the need for multiple languages - https://github.com/travis-ci/travis-ci/issues/4090 was where I ended up). The Trusty setup is setup to be a "mega" build machine and just has everything.

**Comment by zpao** [comment]
For some additional context which I forgot to include initially: I upgraded to Jekyll 3 (#7308)  and at least one of the dependencies now requires Ruby `~> 2.0` in order for us to build the website. https://travis-ci.org/facebook/react/jobs/145969126 is the failed build.

**Pull request merged**